app/models/account.rb

```diff
@@ -43,11 +43,9 @@ class Account < ApplicationRecord
   # @return [Account] A persisted instance of Account
   #
   def self.cache_or_create_by!(user)
-    return unless user.uuid || user.sec_id
+    return unless get_key(user)
```

Comment:
I think the `return` statement has become unreachable code as a result of this change.

Assuming that `user.uuid` & `user.sec_id` are both `nil`:
- `user.uuid || user.sec_id` evaluates falsey
- `get_key(user)` returns "sec:" and evaluates truthy

Reply:
good catch, the original implementation of `get_key` didn't prefix with "sec:"